pkg/registry/core/service/proxy.go

```diff
@@ -71,6 +81,59 @@ func (r *ProxyREST) Connect(ctx genericapirequest.Context, id string, opts runti
 	return newThrottledUpgradeAwareProxyHandler(location, transport, true, false, responder), nil
 }
 
+type redirectProxyHandler struct {
+	ctx         genericapirequest.Context
+	serviceID   string
+	serviceRest *REST
+	pods        *podstore.REST
+	responder   rest.Responder
+	path        string
+}
+
+func (r *redirectProxyHandler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	obj, err := r.serviceRest.Get(r.ctx, r.serviceID, nil)
+	if err != nil {
+		r.responder.Error(fmt.Errorf("error getting service: %v", err))
+		return
+	}
+	svc, ok := obj.(*api.Service)
+	if !ok {
+		r.responder.Error(fmt.Errorf("unexpected type: %v", obj))
+		return
+	}
+	opts := &metainternalversion.ListOptions{
+		LabelSelector: labels.Set(svc.Spec.Selector).AsSelector(),
+	}
+	obj, err = r.pods.List(r.ctx, opts)
+	if err != nil {
+		r.responder.Error(fmt.Errorf("error getting matching pods: %v", err))
+		return
+	}
+	podList, ok := obj.(*api.PodList)
+	if !ok {
+		r.responder.Error(fmt.Errorf("unexpected type: %v", obj))
+		return
+	}
+	if len(podList.Items) == 0 {
+		r.responder.Error(fmt.Errorf("No matching pods found."))
+		return
+	}
+	ix := rand.Intn(len(podList.Items))
+	pod := podList.Items[ix]
+	path := fmt.Sprintf("/apis/v1/namespaces/%s/pods/%s/proxy%s", pod.Namespace, pod.Name, r.path)
```

Comment:
this drops port and protocol info (see logic in service `REST#ResourceLocation`):
```
svcScheme, svcName, portStr, valid := utilnet.SplitSchemeNamePort(id)
```